bundles/org.smarthomej.binding.androiddebugbridge/src/main/java/org/smarthomej/binding/androiddebugbridge/internal/AndroidDebugBridgeHandler.java

```diff
@@ -163,22 +166,43 @@ private void handleCommandInternal(ChannelUID channelUID, Command command)
                 }
             case WAKE_LOCK_CHANNEL:
                 if (command instanceof RefreshType) {
-                    updateState(channelUID, new DecimalType(adbConnection.getPowerWakeLock()));
+                    int wakeLockState = adbConnection.getPowerWakeLock();
+                    int lastWakeLockState = (int) channelLastStateMap.getOrDefault(WAKE_LOCK_CHANNEL, 0);
+                    if (wakeLockState == lastWakeLockState) {
+                        updateState(channelUID, new DecimalType(wakeLockState));
+                    }
+                    channelLastStateMap.put(WAKE_LOCK_CHANNEL, wakeLockState);
```

Comment:
I have to admit, that I do not like these changes. Especially for the wakeLock Channel. I tested it and it really worsened my user experience.

Additionally in my POV debouncing has not to be placed into a Binding. Therefore I created https://github.com/smarthomej/addons/pull/113. A long overdue feature for openHAB.

Please revert these changes. One thing which can be reused is the internal State cache. e.g. the wakeLock is retrieved twice if a user linked the media-control Channel and configured a wakeLock-based "PLAY" detection. We can reorder the refresh requests in [AndroidDebugBridgeHandler# refreshStatus()](https://github.com/smarthomej/addons/blob/abc6a63ed7e4bd83d8ecdb1b5674639d25e59ebe/bundles/org.smarthomej.binding.androiddebugbridge/src/main/java/org/smarthomej/binding/androiddebugbridge/internal/AndroidDebugBridgeHandler.java#L369) to request the wakeLock first, store / cache it internally and reuse it for mediaControl detection.

One could add an `ExpiringCacheMap` for adb calls too.

Reply:
Hi,

for me the double check is only necessary for the hdmi channel. As i know for Miguel it is only necessary for the screen channel. I transfered the double check to all state channels, so maybe this was a bad decision. I will revert it fot the wake lock tomorrow.

br

Reply:
Thanks for looking into it. What do you think about https://github.com/smarthomej/addons/commit/080e29e37e8dee4f39ad4a7e74cb6f62477f3d9b?

Reply:
For me it looks fine, no doubts about this strategy from my side. Thanks for the improvements, too.